### DISCUSSION TITLE
How to enable a package like "centered-cursor-mode" in the minibuffer completion with vertico?

### DISCUSSION THREAD

**Discussion by IceAsteroid** [minad/vertico#465] [Q&A] [answered]
I found it so convenient for looking through lines in a buffer with the cursor and current line fixed in the middle of a buffer inside a window with ```centered-cursor-mode```.

I've also tried to enable the package by adding it to the ```minibuffer-mode-hook```, but it didn't work at all.

And the ```centered-cursor``` mode is already enabled in the minibuffer just like in other buffers, but didn't take effect.

How to possibly make this package take effect when also in the minibuffer? 

I also tried to use the built-in ```scroll-lock-mode``` package, though it's not perfect, but the minibuffer then is frozen to not be able to select candidates by scrolling with ```C-p``` or ```C-n```. So it's not compatible in minibuffer

**Comment by minad** [comment, marked as answer]
Such modes do not work with Vertico, due to the way the Vertico UI is implemented via overlays. You can try this instead:

~~~ elisp
(setq vertico-scroll-margin (/ vertico-count 2))
~~~

**Reply by IceAsteroid** [reply]
It works fantastically. Thank you! Sorry for the ask, I've searched but haven't known it was for this.

However, when the ```vertico-count``` variable is equal or greater than 31 in my environment for this case, it would result like this:

![vertico-balanced-scroll](https://github.com/minad/vertico/assets/90975914/0173c9a4-7fe9-4600-8ed3-b7cab013d23e)

It would start shifting from the middle of the minibuffer. From the above image, it's set to 40. With smaller numbers, it'll be less to shift

But it's trivial, since usually there's no need for candidates listed over 31, and it's still not that bad

**Reply by minad** [reply]
> However, when the vertico-count variable is equal or greater than 31 in my environment for this case, it would result like this:

I think the problem is that your vertico-count in this case is larger than the maximal mini window size. Vertico assumes that it can grow the window to vertico-count lines.

**Reply by IceAsteroid** [reply]
That was indeed the cause. Just set a smaller number, it's fine now, thank you!
